sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/implementation/AmqpReceiveLinkProcessor.java

```diff
@@ -177,42 +202,45 @@ public void onNext(AmqpReceiveLink next) {
                                 LinkErrorContext errorContext = (LinkErrorContext) amqpException.getContext();
                                 if (currentLink != null
                                     && !currentLink.getLinkName().equals(errorContext.getTrackingId())) {
-                                    logger.info("EntityPath[{}]: Link lost signal received for a link "
-                                        + "that is not current. Ignoring the error. Current link {}, link lost {}",
-                                        entityPath, linkName, errorContext.getTrackingId());
+                                    logger.info("linkName[{}] entityPath[{}] trackingId[{}] Link lost signal received"
+                                            + " for a link that is not current. Ignoring the error.",
+                                        linkName, entityPath, errorContext.getTrackingId());
                                     return;
                                 }
                             }
                         }
+
                         currentLink = null;
-                        logger.warning("linkName[{}] entityPath[{}]. Error occurred in link.", linkName, entityPath);
                         onError(error);
                     },
                     () -> {
                         if (parentConnection.isDisposed() || isTerminated()
                             || UPSTREAM.get(this) == Operators.cancelledSubscription()) {
-                            logger.info("Terminal state reached. Disposing of link processor.");
+                            logger.info("linkName[{}] entityPath[{}] Terminal state reached. Disposing of link "
+                                + "processor.", linkName, entityPath);
+
                             dispose();
                         } else {
-                            logger.info("Receive link endpoint states are closed. Requesting another.");
+                            logger.info("linkName[{}] entityPath[{}] Receive link endpoint states are closed. "
+                                + "Requesting another.", linkName, entityPath);
+
                             final AmqpReceiveLink existing = currentLink;
                             currentLink = null;
+                            currentLinkName = null;
 
-                            if (existing != null) {
-                                existing.dispose();
-                            }
+                            disposeReceiver(existing);
                             requestUpstream();
                         }
                     }),
-                next.receive().subscribe(message -> {
-                    messageQueue.add(message);
-                    drain();
-                }));
+                next.receive()
+                    .onBackpressureBuffer(maxQueueSize, BufferOverflowStrategy.ERROR)
+                    .subscribe(message -> {
```

Comment:
Should this subscribe have an error  handler to dispose the link if there's an error?

Reply:
No because the underlying deliverySink this fetches from never outputs an onError, it will do that via the endpointStates